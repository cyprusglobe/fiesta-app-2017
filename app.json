{
  "name": "FiestaApp2017",
  "displayName": "FiestaApp2017"
}