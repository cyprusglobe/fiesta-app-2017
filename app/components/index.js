export { default as InstagramPhotos } from './InstagramPhotos';
export { default as InstagramPhotoModal } from './InstagramPhotoModal';
export { default as InstagramPhoto } from './InstagramPhoto';
export { default as ProgressBar } from './ProgressBar';
export { default as StartHelper } from './StartHelper';

export { default as MenuItem } from './MenuItem';
export { default as ProfileItem } from './ProfileItem';
export { default as PilotItem } from './PilotItem'
export { default as BalloonItem } from './BalloonItem';

export { default as MenuButton } from './MenuButton';
export { default as Menu } from './Menu';
export { default as CountDown } from './CountDown';
export { default as Weather } from './Weather';

export { default as CornerLabel } from './CornerLabel';
