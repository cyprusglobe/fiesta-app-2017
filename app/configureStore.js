import Reactotron from 'reactotron-react-native';

import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';

import rootReducer from './modules';

function configureStore(initialState) {
  // const devTools = require('remote-redux-devtools');
  /**
   * Create store with remote-devtools and logger middleware. Do it only
   * in development to reduce performance issues.
   */
  if (__DEV__) {
    // const createLogger = require('redux-logger')
    // const logger = createLogger()

    const store = Reactotron.createStore(
      rootReducer,
      {},
      compose(applyMiddleware(thunkMiddleware))
    );

    return store;
  } else {
    // const finalCreateStore = compose(applyMiddleware(thunkMiddleware), devTools())(createStore)
    const finalCreateStore = applyMiddleware(thunkMiddleware)(createStore);
    const store = finalCreateStore(rootReducer, initialState);

    return store;
  }
}

export default configureStore;
