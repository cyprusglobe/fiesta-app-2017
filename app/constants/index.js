/**
 * @flow
 */

const IMAGE_URL =
  'https://registration.balloonfiesta.com/hotair/balloon/image/';
const TIME_FORMAT = 'h:mma';

export { IMAGE_URL, TIME_FORMAT };
