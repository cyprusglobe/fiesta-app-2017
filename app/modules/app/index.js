/**
 * @flow
 */

import { initializeUi } from '../ui';
import { NetInfo, AppState } from 'react-native';

/**
 * Constants
 */
export const INITIALIZED = 'app/INITIALIZED';
export const CHANGE_NET_STATUS = 'app/CHANGE_NET_STATUS';
export const CHANGE_APP_STATE = 'app/CHANGE_APP_STATE';

/**
 * Action Creators
 */

export function init(startAppWithScreen, navigator) {
  return async (dispatch, getState) => {
    dispatch(setupAppStatusListener());
    dispatch(setupNetStatusListener());

    const navigate = startAppWithScreen
      ? startAppWithScreen
      : navigator.resetTo;

    // dispatch({ type: INITIALIZED, date: { time: new Date().toString() } })
    try {
      const netStatus = await NetInfo.fetch();

      if (netStatus === 'none' || netStatus === 'NONE') {
        navigate({ screen: 'bf.Pilots' });
        return;
      }
    } catch (error) {
      dispatch({ type: INITIALIZED, error: error.message });
      navigate({ screen: 'bf.Home' });
    }
    navigate({ screen: 'bf.Home' });
  };
}

function setupNetStatusListener() {
  return dispatch => {
    NetInfo.isConnected.addEventListener('change', async status => {
      dispatch({ type: CHANGE_NET_STATUS, payload: status });
    });
  };
}

function setupAppStatusListener() {
  return async (dispatch, getState) => {
    AppState.addEventListener('change', async status => {
      try {
        dispatch({ type: CHANGE_APP_STATE, payload: status });
        const netStatus = getState().app.status;
        if (!netStatus) {
          return;
        }
      } catch (error) {}
    });
  };
}

/**
 * Reducer
 */

const initialState = {
  online: false,
  appState: null,
  fayeConnected: false,
};

export default function app(state = initialState, action) {
  switch (action.type) {
    case INITIALIZED:
      return {
        ...state,
        error: app.payload,
      };
    case CHANGE_NET_STATUS:
      return {
        ...state,
        online: action.payload,
      };
    case CHANGE_APP_STATE:
      return {
        ...state,
        appState: action.payload,
      };
    default:
      return state;
  }
}
