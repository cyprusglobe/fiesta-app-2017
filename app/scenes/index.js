/**
 * @flow
 */

export { default as Menu } from './Menu';

export { default as Welcome } from './Welcome';
export { default as Home } from './Home';
export { default as Instagram } from './Instagram';
export { default as Search } from './Search';

export { default as Balloons } from './Balloons';
export { default as Balloon } from './Balloon';

export { default as Pilots } from './Pilots';
export { default as Pilot } from './Pilot';

export { default as Notifications } from './Notifications'
