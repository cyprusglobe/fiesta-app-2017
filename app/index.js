/**
 * @flow
 */

import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from './modules';
import configureStore from './configureStore';
import Application from './scenes';

const store = configureStore();

const app = new Application(store, Provider);

app.run();
