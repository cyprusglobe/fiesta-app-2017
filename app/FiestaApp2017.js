/**
 * @flow
 */

import React, { Component } from 'react'
import {
  AppRegistry,
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Image,
} from 'react-native'
import codePush from 'react-native-code-push'
import Icon from 'react-native-vector-icons/Ionicons'
import { StackNavigator } from 'react-navigation'

import syncEnchancer from './utils/syncEnhancer'
import { Home, Settings } from './scenes'

import { Navigation } from 'react-native-navigation';

function registerScreens() {
  Navigation.registerComponent('example.Home', () => Home)
  Navigation.registerComponent('example.Settings', () => Settings)
}

registerScreens()
// start the app
syncEnchancer(Navigation.startSingleScreenApp({
  screen: {
    screen: 'example.Home', // unique ID registered with Navigation.registerScreen
    // title: 'Home', // title of the screen as appears in the nav bar (optional)
    navigatorStyle: {}, // override the navigator style for the screen, see "Styling the navigator" below (optional)
    navigatorButtons: {} // override the nav buttons for the screen, see "Adding buttons to the navigator" below (optional)
  },
}));
