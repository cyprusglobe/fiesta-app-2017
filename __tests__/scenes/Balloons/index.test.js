/**
 * @flow
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { Balloons } from '../../../app/scenes';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Balloons />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
