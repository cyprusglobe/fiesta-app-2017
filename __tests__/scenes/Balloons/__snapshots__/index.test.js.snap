// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#DFDFDF",
      "height": "100%",
      "paddingTop": 64,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": 125,
            "justifyContent": "center",
            "marginVertical": 10,
          },
        ]
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</View>
`;
